src/amuse/units/trigo.py

```diff
@@ -9,7 +9,10 @@
 arcsin = lambda x: numpy.arcsin(x) | rad
 arccos = lambda x: numpy.arccos(x) | rad
 arctan = lambda x: numpy.arctan(x) | rad
-arctan2 = lambda x, y: numpy.arctan2(x, y) | rad
+arctan2 = lambda x, y: numpy.arctan2(
+    x.value_in(x.unit) if isinstance(x, quantities.Quantity) else x,
+    y.value_in(x.unit) if isinstance(x, quantities.Quantity) else y,
```

Comment:
Note that this explicitly checks the unit of `x` in both cases, and also whether `x` is a `Quantity`. This is because `x` and `y` need to be treated in the same way, otherwise they might have different units.